### PR TITLE
Delete Step in Integration Editor

### LINKED ISSUE #259: Editor - implement the delete step action and confirmation dialog
## This is a...

<pre><code>
[x] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
The trash icon needs to be hooked up for steps.  In all cases the trash icon should prompt the user if they really want to remove the step.  

For steps at the start and end of the flow the step should be removed but the user should then be brought to the step select page for that position and be required to set a step/connection and configure it.

For steps in the middle positions after confirmation the step is simply removed from the step array.

Special steps such as the API provider start/finish connections and the conditional flow start/finish connections cannot be removed, so the trash icon should be either hidden or disabled.  The angular app hides them.

### PR DESCRIPTION
**Pull request by kahboom**
Delete steps in the Edit Integration page.

fix #259

### CODE REVIEW

File: app/ui-react/syndesis/src/modules/integrations/components/IntegrationEditorStepAdder.tsx [hunk 1]
```diff
@@ -45,74 +60,145 @@ export interface IIntegrationEditorStepAdderProps {
...
+                      'integrations:editor:confirmDeleteStepDialogBody'
+                    )}
+                    i18nTitle={t(
+                      'integrations:editor:confirmDeleteStepDialogTitle'
+                    )}
+                    showDialog={this.state.showDeleteDialog}
+                    onCancel={this.closeDeleteDialog}
+                    onConfirm={() => {
+                      this.handleDeleteConfirm();
+                      removeStepFromFlow(
```
Comment (gashcrumb) [inline app/ui-react/syndesis/src/modules/integrations/components/IntegrationEditorStepAdder.tsx]: yeah, this here function returns a copy of the integration object with that step removed from the indicated flow.  This is where I guess we need to figure out how to get the page to see the return value of this function.

Comment (kahboom) [reply app/ui-react/syndesis/src/modules/integrations/components/IntegrationEditorStepAdder.tsx]: Just to confirm then, what you are saying is that if I set that function as a local var, it should return one less step than the original integration, correct? But I do this:
```
onConfirm={() => {
                      console.log('oldInt: ' + JSON.stringify(this.props.integration));
                      this.handleDeleteConfirm();
                      const newInt = removeStepFromFlow(
                        this.props.integration!,
                        this.props.flowId!,
                        this.state.position!
                      );
                      console.log('newInt: ' + JSON.stringify(newInt));
                    }}
```
And the old integration's `steps` array count is the same as the new one. I would expect one less step for the returned copy of the integration object. I tested that the data passed is correct, so I want to make sure the function is working as expected.

Comment (kahboom) [reply app/ui-react/syndesis/src/modules/integrations/components/IntegrationEditorStepAdder.tsx]: Got it, I tested both `const steps = [...flow!.steps!];` (current implementation) and `const steps = getSteps(integration, flowId);` (my changes) for the steps, and it seems to work with my changes. Moving to the next step now (updating the page).

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/1532/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Pull request merged**